### PR TITLE
Update screenshotter

### PR DESCRIPTION
**Pull request by dennari**
The new screenshotter image needs to be pulled from Lucify's private docker registry. This requires logging in with the docker client:

```bash
assume_admin xxx
ecs-updater -s login
docker-compose up
```

### CODE REVIEW

File: src/screenshot/screenshot-module.ts [hunk 1]
```diff
@@ -72,35 +63,38 @@ export default class ScreenshotModule {
...
   /*
    * Take a screenshot for given projectId and deploymentId
    */
   public async takeScreenshot(projectId: number, deploymentId: number, shortId: string) {
     const url = sprintf(this.urlPattern, `${shortId}-${projectId}-${deploymentId}`);
+    const dest = this.getRemoteDest(projectId, deploymentId);
+    const options: PageresOptions = {
+      filename: this.getScreenshotFilename(projectId, deploymentId).replace(/\.[^.]+$/, ''),
+      delay: 5,
+      format: 'jpg',
```
Comment (vsaarinen) [inline src/screenshot/screenshot-module.ts]: Isn't PNG a better format for screenshots?

File: src/screenshot/screenshot-module.ts [hunk 2]
```diff
@@ -72,35 +63,38 @@ export default class ScreenshotModule {
...
   }
 
   /*
    * Take a screenshot for given projectId and deploymentId
    */
   public async takeScreenshot(projectId: number, deploymentId: number, shortId: string) {
     const url = sprintf(this.urlPattern, `${shortId}-${projectId}-${deploymentId}`);
+    const dest = this.getRemoteDest(projectId, deploymentId);
+    const options: PageresOptions = {
+      filename: this.getScreenshotFilename(projectId, deploymentId).replace(/\.[^.]+$/, ''),
```
Comment (vsaarinen) [inline src/screenshot/screenshot-module.ts]: I :heart: regexps, but maybe take out this one into a separate `stripFileExtension` method to make it more explicit what it's doing?

File: tsconfig.json [hunk 3]
```diff
@@ -7,11 +7,13 @@
     "outDir": "./dist/",
     "sourceMap": true,
     "noImplicitAny": true,
+    "noImplicitThis": false,
```
Comment (vsaarinen) [inline tsconfig.json]: Do thing break if this is set to `true`?

File: src/screenshot/screenshot-module.ts (hunk 1 above)
Comment (dennari) [inline src/screenshot/screenshot-module.ts]: It's not so clear cut. We definitely get smaller files with the lossy JPGs and the way the screenshots are currently used by the UI makes other factors quite unimportant. If at some point we'd like to have HR images where text legibility is crucial, then maybe PNG.

File: tsconfig.json (hunk 3 above)
Comment (dennari) [inline tsconfig.json]: I couldn't find a workaround for this one Hapi handler extension function, which uses `this` inside a function with purpose.

File: docker-compose.yml [hunk 4]
```diff
@@ -165,7 +165,7 @@ services:
       - DOCKER_VOLUMES=/cache
 
   screenshotter:
-    image: lucify/screenshotter
+    image: 366857391347.dkr.ecr.eu-west-1.amazonaws.com/screenshotter:63e5b0
```
Comment (juhoojala) [inline docker-compose.yml]: The plain docker-compose should use public images, and the other ones should be in docker-compose-ecr.yml. 

I changed this.

**Comment by juhoojala** [comment]
:+1:

**Pull request merged**